src/util/Iterators.h

```diff
@@ -400,10 +422,10 @@ class RangeToInputRangeFromGet
 // A simple type-erased input range (that is, one class for *any* input range
 // with the given `ValueType`). It internally uses the `InputRangeOptionalMixin`
 // from above as an implementation detail.
-template <typename ValueType>
+template <typename ValueType, typename DetailsType = NoDetails>
```

Comment:
Maybe we can get rid of the InputRangeTypeErasedWithDetails using this approach + the mixin, then the interfaces is the easiest I think. 
(You can start with all the other changes, and then maybe we can put this update of the range facilities in a smaller splinter PR, because other PRs also depend on it.

Reply:
fully agree